homeassistant/components/wemo/__init__.py

```diff
@@ -33,13 +36,24 @@
 def coerce_host_port(value):
     """Validate that provided value is either just host or host:port.
 
+    If a hostname is provided, resolve it to an IP address before proceeding.
+
     Returns (host, None) or (host, port) respectively.
     """
     host, _, port = value.partition(":")
 
     if not host:
         raise vol.Invalid("host cannot be empty")
 
+    # Force hostnames into IP addresses for PyWemo compatibility
+    try:
+        ip_address(host)
+    except ValueError:
+        # This was not an IP; try to resolve it
+        resolved_ip = gethostbyname(host)
```

Comment:
Shouldn't this be a concern for the Python package that provides the communication? (`pywemo` in this case)

Reply:
The package recommends against hostnames anyways for other reasons, and implementing the resolver logic there would be harder (more disparate functions) than just ensuring that HASS is sending it IPs only. I debated both solutions, but this one seems like the easiest one.

Reply:
Between this and the issue with the event loop, I'd agree with you here. I'll focus on the PyWemi side of this equation and close this out.